xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/store/hibernate/internal/MySQLHibernateAdapter.java

```diff
@@ -0,0 +1,172 @@
+                session.getTransaction().begin();
+                query.executeUpdate();
+                session.getTransaction().commit();
+            }
+        }
+    }
+
+    /**
+     * @param tableName the name of the table to modify
+     * @param compressed true if the table should be compressed
+     * @param rowFormats the current row formats
+     * @param session the session in which to execute the query
+     * @return the SQL statement to execute if the table is not already compressed
+     * @throws HibernateStoreException when failing to check if the table is already compressed
+     */
+    public String getAlterRowFormatString(String tableName, boolean compressed, Map<String, String> rowFormats,
+        Session session) throws HibernateStoreException
+    {
+        String rowFormat = rowFormats.get(tableName);
+
+        this.logger.debug("Row format for table [{}]: {}", tableName, rowFormat);
+
+        String expectedRowFormat = compressed ? getCompressedRowFormat() : getDefaultRowFormat();
+
+        if (!StringUtils.equalsIgnoreCase(rowFormat, expectedRowFormat)) {
+            return "ALTER TABLE " + tableName + " ROW_FORMAT=" + expectedRowFormat;
+        }
+
+        return null;
+    }
+
+    /**
+     * @param session the session in which to execute the query
+     * @return the row format of each table in the current database
+     */
+    public Map<String, String> getRowFormats(Session session)
+    {
+        String currentDatabase = getDatabaseFromWikiName();
+        NativeQuery<Object[]> query = session.createNativeQuery("SELECT DISTINCT NAME, ROW_FORMAT FROM "
+            + getInformationShemaInnoDBTables() + " WHERE LOWER(NAME) LIKE '" + currentDatabase.toLowerCase() + "/%'");
+
+        List<Object[]> results = query.list();
+
+        Map<String, String> rowFormats = new HashMap<>(results.size());
+        for (Object[] entry : results) {
+            String completeName = (String) entry[0];
+            String tableName = completeName.substring(completeName.indexOf('/') + 1);
+            rowFormats.put(tableName, (String) entry[1]);
+        }
+
+        return rowFormats;
+    }
+
+    /**
+     * @return the table where to find the InnoDB tables metadata
+     */
+    public String getInformationShemaInnoDBTables()
+    {
+        return "INFORMATION_SCHEMA.INNODB_SYS_TABLES";
+    }
+
+    /**
+     * @return the row format to use by default
+     */
+    public String getDefaultRowFormat()
+    {
+        return "Dynamic";
+    }
+
+    /**
+     * @return the row format to use for compressed tables
+     */
+    public String getCompressedRowFormat()
+    {
+        return "Compressed";
+    }
+
+    @Override
+    public boolean isCompressionAllowed()
+    {
+        Optional<Boolean> compressionAllowed = getCompressionAllowedConfiguration();
+
+        if (compressionAllowed.isPresent()) {
+            return compressionAllowed.get();
+        }
+
+        return true;
+    }
```

Comment:
Shorter equivalent
```suggestion
    @Override
    public boolean isCompressionAllowed()
    {
        return getCompressionAllowedConfiguration().orElse(true);
    }
```

Reply:
Right, I knew there was something like this but did not dig. Thanks :)